ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/AnnotationRule.kt

```diff
@@ -140,6 +144,67 @@ class AnnotationRule : Rule("annotation") {
                 }
             }
         }
+
+        // Check to make sure no trailing line breaks between annotation and object
+        val lineNumber = node.lineNumber()
+        val next = node.nextSiblingWithAtLeastOneOf(
+            {
+                !it.isWhiteSpace() &&
+                    it.textLength > 0 &&
+                    !(it.isPartOfComment() && it.lineNumber() == lineNumber) &&
+                    !it.isPartOf(FILE_ANNOTATION_LIST)
+            },
+            {
+                val s = it.text
+                // Ensure at least one occurrence of two line breaks
+                s.indexOf("\n") != s.lastIndexOf("\n")
+            }
+        )
+        val nextLineNumber = next?.lineNumber()
+        if (lineNumber != null && nextLineNumber != null) {
+            val diff = nextLineNumber - lineNumber
+            // Ensure declaration is not on the same line, there is a line break in between, and it is not an
+            // annotation we explicitly want to have a line break between
+            if (lineNumber != nextLineNumber && diff > 1 && !node.text.contains("@file")) {
```

Comment:
I think `diff > 1` is sufficient here right?

also not sure about the check for `@file` presence - aren't you doing this already above with `!it.isPartOf(FILE_ANNOTATION_LIST)`?